clang/lib/Driver/ToolChains/Arch/AArch64.cpp

```diff
@@ -81,6 +81,15 @@ static bool DecodeAArch64Features(const Driver &D, StringRef text,
     else
       return false;
 
+    // +jsconv and +complxnum implies +neon and +fp-armv8
```

Comment:
I believe this ideally would not be in the driver, as it does not apply to target attributes, only -march options.

Reply:
The dependencies between these new target attributes and others are also specified in llvm/include/llvm/TargetParser/AArch64TargetParser.h. And we want to support having nofp and nosimd disable jsconv and complxnum. This piece of code makes the driver pass the correct -target-feature flags to the backend.

Reply:
According to the latest Arm ARM (https://developer.arm.com/documentation/ddi0487/ja/?lang=en) the architecrtural extensions `FEAT_FCMA` and `FEAT_JSCVT` are mandatory in Armv8.3-a and are not optional in any architecture version.

For such features, our convention is to not expose them as command-line options.

Reply:
@momchil-velikov Thank you for your answer. It is strange that the wording in the Arm ARM is so strict. It is well understood that an Armv8.*x*-A processor can implement any feature from the next *.x* generation (see [Understanding the Armv8.x extensions](https://developer.arm.com/-/media/Arm%20Developer%20Community/PDF/Learn%20the%20Architecture/Understanding%20the%20Armv8.x%20extensions.pdf)).

The rationale for this change is to enable the use of these ISA features on an Armv8.2-a processor (such as TSV110 and other similar chips), by making the command-line options available.

Reply:
@bryanpkc Good point. The same is said also in Arm ARM, section A2.1.1.

There is already a description  at https://github.com/llvm/llvm-project/blob/ba4d36951f1deaf24dce526900a2d6dbdefa377b/llvm/lib/Target/AArch64/AArch64.td#L1513

Shouldn't it include both  `FeatureJS` and `FeatureComplxNum` ?

Once you have that, I would expect `+nosimd` to disable FEAT_FCMA and `+nofp` to disable both `FEAT_FCMA` and `FEAT_JSCVT`.

Reply:
Ah yes, this PR is missing a modification in the .td file. @Qi-Hu Could youu please try that?

Reply:
Sure, it has been added.

Reply:
I have made some changes and included AEK_JSCVT and AEK_FCMA in Armv8.3-A. Could you please review it again? @davemgreen @momchil-velikov Thanks!